packages/cactus-common/src/main/typescript/logging/log-helper.ts

```diff
@@ -0,0 +1,100 @@
+    let exceptionHandled = false;
+
+    // 1st need to check that exception is not null or undefined before trying to access the wanted stack information
+    if (exception) {
+      if (exception instanceof RuntimeError) {
+        // handling RuntimeError stack inclusive nested / cascaded stacks
+        stack = this.safeJsonStringify(exception);
+        exceptionHandled = true;
+      }
+
+      if (!exceptionHandled && typeof exception === "object") {
+        // 2nd need to check if a stack property is available
+        if (Object.hasOwnProperty.call(exception, "stack")) {
+          // 3rd check if the stack property is already of type string
+          if (
+            typeof (exception as Record<string, unknown>).stack === "string"
+          ) {
+            stack = (exception as { stack: string }).stack;
+          } else {
+            // need to stringify stack information first
+            stack = this.safeJsonStringify(
+              (exception as { stack: unknown }).stack,
+              invalidStack,
+            );
+          }
+        }
+      }
+    }
+    return stack;
+  }
+
+  public static getExceptionMessage(exception: unknown): string {
+    // handle unknown exception input
+    const defaultMessage = "NO_MESSAGE_INCLUDED_IN_EXCEPTION";
+    const invalidException = "INVALID_EXCEPTION";
+    const invalidMessage = "INVALID_EXCEPTION_MESSAGE";
+    const customExceptionPrefix = "A CUSTOM EXCEPTION WAS THROWN: ";
+    let message = defaultMessage;
+    let exceptionHandled = false;
+
+    // 1st need to check that exception is not null or undefined before trying to access the wanted message information
+    if (exception) {
+      if (typeof exception === "object") {
+        // 2nd need to check if a message property is available
+        if (Object.hasOwnProperty.call(exception, "message")) {
+          // 3rd check if the message property is already of type string
+          if (
+            typeof (exception as Record<string, unknown>).message === "string"
+          ) {
+            message = (exception as { message: string }).message;
+          } else {
+            // need to stringify message information first
+            message = this.safeJsonStringify(
+              (exception as { message: unknown }).message,
+              invalidMessage,
+            );
+          }
+          exceptionHandled = true;
+        }
+      }
+
+      // handling of custom exceptions
+      if (!exceptionHandled) {
+        // check if thrown custom exception is a string type only -> directly use it as exception message
+        if (typeof exception === "string") {
+          message = exception;
+        } else {
+          // custom exception is of a different type -> need to stringify it
+          message =
+            customExceptionPrefix &&
+            this.safeJsonStringify(exception, invalidException);
+        }
+      }
+    }
+    return message;
+  }
+
+  private static safeJsonStringify(
```

Comment:
@petermetz: You mentioned some time ago that the cactus-common module is a critical one as it is getting used nearly everywhere. Therefore I was a bit conservative and not including any additional external libraries into this. But for me definitely fine to integrate the "fast-safe-stringify" library.

Reply:
> @petermetz: You mentioned some time ago that the cactus-common module is a critical one as it is getting used nearly everywhere. Therefore I was a bit conservative and not including any additional external libraries into this. But for me definitely fine to integrate the "fast-safe-stringify" library.

@m-courtin Yeah, I also did not make that recommendation easily for the same reason. I did take a look at the package size before and while the 40 KBs it takes up in space is not ideal (I feel like it could be done in a few KBs) it's also not that bad[1] for something as crucial as this (I have nightmares about debugging production issues with log messages of `[Object object]` in them or with crashes within the catch blocks, etc.)

[1]: Unfortunately most packages nowadays are well above a megabyte even when their code itself seems very small.